tools/pre-swap-checker/contracts/PreSwapChecker.sol

```diff
@@ -36,87 +37,96 @@ contract PreSwapChecker {
         errors[i] = swapErrors[i];
       }
     }
+    if (delegateErrorCount > 0) {
 
-    // signature must be filled in order to use the Delegate
-    if (order.signature.v == 0) {
-      errors[errorCount] = "SIGNATURE_MUST_BE_SENT";
-      errorCount++;
+      // copies over errors from checkSwapSwap to be outputted
+      for (uint256 i = 0; i < delegateErrorCount; i++) {
+        errors[i + errorCount] = delegateErrors[i];
+      }
+      errorCount = swapErrorCount + delegateErrorCount;
     }
 
-    // check that the sender.wallet == tradewallet
-    if (order.sender.wallet != delegate.tradeWallet()) {
-      errors[errorCount] = "SENDER_WALLET_INVALID";
-      errorCount++;
-    }
+    // Check valid token registry handler for sender
+    if (hasValidKind(order.sender.kind, swap) && order.sender.kind == ERC20_INTERFACE_ID) {
+      // Check the order sender balance and allowance
+      if (!hasBalance(order.sender)) {
+        errors[errorCount] = "SENDER_BALANCE";
+        errorCount++;
+      }
 
-    // ensure signer kind is ERC20
-    if (order.signer.kind != ERC20_INTERFACE_ID) {
-      errors[errorCount] = "SIGNER_KIND_MUST_BE_ERC20";
-      errorCount++;
+      // Check their approval
+      if (!isApproved(order.sender, swap)) {
+        errors[errorCount] = "SENDER_ALLOWANCE";
+        errorCount++;
+      }
     }
 
-    // ensure sender kind is ERC20
-    if (order.sender.kind != ERC20_INTERFACE_ID) {
-      errors[errorCount] = "SENDER_KIND_MUST_BE_ERC20";
-      errorCount++;
-    }
 
-    // ensure that token pair is active with non-zero maxSenderAmount
-    if (rule.maxSenderAmount == 0) {
-      errors[errorCount] = "TOKEN_PAIR_INACTIVE";
-      errorCount++;
-    }
+     // Check valid token registry handler for signer
+    if (hasValidKind(order.signer.kind, swap) && order.signer.kind == ERC20_INTERFACE_ID) {
+      if (order.signer.token != address(wethContract)) {
+        // Check the order signer token balance
+        if (!hasBalance(order.signer)) {
+          errors[errorCount] = "SIGNER_BALANCE";
+          errorCount++;
+        }
+      }
 
-    if (order.sender.amount > rule.maxSenderAmount) {
-      errors[errorCount] = "ORDER_AMOUNT_EXCEEDS_MAX";
-      errorCount++;
+      // Check their approval
+      if (!isApproved(order.signer, swap)) {
+        errors[errorCount] = "SIGNER_ALLOWANCE";
+        errorCount++;
+      }
     }
 
-    // calls the getSenderSize quote to determine how much needs to be paid
-    uint256 senderAmount = delegate.getSenderSideQuote(order.signer.amount, order.signer.token, order.sender.token);
-    if (senderAmount == 0) {
-      errors[errorCount] = "DELEGATE_UNABLE_TO_PRICE";
-      errorCount++;
-    } else if (order.sender.amount > senderAmount) {
-      errors[errorCount] = "PRICE_INVALID";
-      errorCount++;
+    // if signer has WETH token, ensure sufficient ETH balance
+    if (order.signer.token == address(wethContract)) {
```

Comment:
Can't this be an `else` with line 93 `if (order.signer.token != address(wethContract)) {` instead of a new if statement?